### PR TITLE
Move to Sphinx 4 and replace cairosvgconverter

### PR DESCRIPTION
**Pull request by ktbarrett**
Removes the dependency restriction from Sphinx, so we can move to Sphinx 4. The implementation of `cairosvgconvert` in the repo stopped working with Sphinx 4 (due to deprecations). So instead we are adding a dependency on an contrib library that does the same thing and works with Sphinx 4.

### CODE REVIEW

File: documentation/requirements.txt [hunk 1]
```diff
@@ -1,10 +1,11 @@
 # Python dependencies to build the documentation
 
 # != 3.5.0: https://github.com/sphinx-doc/sphinx/issues/8880
-Sphinx ~= 3.1, != 3.5.0
+Sphinx != 3.5.0
 breathe
```
Comment (eric-wieser) [inline documentation/requirements.txt:5]: It's probably a good idea to pin to `~= 4.0` just so we don't end up with immediately breakage in random PRs when sphinx 5 is released.

File: documentation/requirements.txt [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # Python dependencies to build the documentation
 
 # != 3.5.0: https://github.com/sphinx-doc/sphinx/issues/8880
-Sphinx ~= 3.1, != 3.5.0
+Sphinx != 3.5.0
```
Comment (ktbarrett) [inline documentation/requirements.txt:4]:
```suggestion
Sphinx ~= 4.0 != 3.5.0
```

File: documentation/requirements.txt [hunk 3]
```diff
@@ -1,10 +1,11 @@
 # Python dependencies to build the documentation
 
 # != 3.5.0: https://github.com/sphinx-doc/sphinx/issues/8880
-Sphinx ~= 3.1, != 3.5.0
+Sphinx ~= 4.0 != 3.5.0
```
Comment (ktbarrett) [inline documentation/requirements.txt:4]:
```suggestion
Sphinx ~= 4.0, != 3.5.0
```

File: documentation/requirements.txt [hunk 4]
```diff
@@ -1,10 +1,11 @@
 # Python dependencies to build the documentation
 
 # != 3.5.0: https://github.com/sphinx-doc/sphinx/issues/8880
-Sphinx ~= 3.1, != 3.5.0
+Sphinx ~= 4.0, != 3.5.0
 breathe
 sphinx-rtd-theme
 cairosvg
+sphinxcontrib-svg2pdfconverter
```
Comment (imphil) [inline documentation/requirements.txt:8]: `sphinxcontrib-svg2pdfconverter[CairoSVG]`

And remove the cairosvg dependency above.

File: documentation/requirements.txt [hunk 5]
```diff
@@ -1,10 +1,11 @@
 # Python dependencies to build the documentation
 
 # != 3.5.0: https://github.com/sphinx-doc/sphinx/issues/8880
-Sphinx ~= 3.1, != 3.5.0
+Sphinx ~= 4.0, != 3.5.0
```
Comment (imphil) [inline documentation/requirements.txt:4]: You can remove the `!= 3.5.0` part, you're already restricting to at least 4.0 with the previous clause. (The comma is an AND here.)

**Review by imphil — APPROVED** [review]
LGTM, thanks!

File: documentation/requirements.txt (hunk 5 above)
Comment (eric-wieser) [reply documentation/requirements.txt:4]:
```suggestion
Sphinx ~= 4.0
```

**Review by eric-wieser — APPROVED** [review]

File: documentation/requirements.txt [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
 breathe
 sphinx-rtd-theme
 cairosvg
```
Comment (eric-wieser) [inline documentation/requirements.txt:7]:
```suggestion
```

File: documentation/requirements.txt (hunk 4 above)
Comment (eric-wieser) [reply documentation/requirements.txt:8]:
```suggestion
sphinxcontrib-svg2pdfconverter[CairoSVG]
```

**Pull request merged**